com.ibm.streamsx.topology/opt/python/packages/streamsx/topology/tester.py

```diff
@@ -289,14 +301,15 @@ def _distributed_wait_for_result(self):
         self.result['submission_result'] = self.submission_result
         if self.local_check is not None:
             self._local_thread.join()
-            if self.local_check_exception is not None:
-                raise self.local_check_exception
+        cc._canceljob(self.result)
```

Comment:
The cc._complete is cancelling the job before the `local_check` function completed.  So I have pulled the job cancel logic out and called it after the `local_check` thread has completed.